location/motif.py

```diff
@@ -833,6 +833,129 @@ def compute_nodes(df,
     return df, nodes
 
 
+def generate_motifs(data,
+                    nodes,
+                    sr_col='stay_region',
+                    round_trip=True,
+                    insert_home=True,
+                    home=None,
+                    dayofweek=[0, 1, 2, 3, 4],
+                    trav_dist_th=50000,
+                    valid_timeslot_th=8):
+    """
+    Generate moitfs for given data.
+
+    Parameters:
+    -----------
+
+    data: DataFrame
+        Location data.
+
+    nodes: tuple
+        Nodes generated by generate_daily_nodes().
+
+    sr_col: str
+        Column name for stay region.
+        Default is 'stay_region'.
+
+    round_trip: bool
+        Whether to consider only days that start and end at the same date.
+        Default is True.
+
+    insert_home: bool
+        Whether to insert home location if the first timeslot of daily
+        data is missing. Default is True.
+
+    home: str
+        Home location in geohash form.
+        Default is None. In this case, home locatoin is approximated using
+        user location data.
+
+    dayofweek: list of integers
+        Which days in a week to consider, [0,1,2,3,4,5,6] for Mon, Tue, Wed,
+        Thurs, Fri, Sat, and Sun. Defualt is weekdays.
+
+    trav_dist_th: int
```

Comment:
I am not sure if we should do this sort of filtering here. It makes more sense to filter out questionable points early in the process. In this phase of analysis, we should have only valid points.